core/src/main/scala/org/apache/spark/deploy/master/Master.scala

```diff
@@ -563,32 +570,48 @@ private[master] class Master(
       app: ApplicationInfo,
       usableWorkers: Array[WorkerInfo],
       spreadOutApps: Boolean): Array[Int] = {
-    // If the number of cores per executor is not specified, then we can just schedule
-    // 1 core at a time since we expect a single executor to be launched on each worker
-    val coresPerExecutor = app.desc.coresPerExecutor.getOrElse(1)
+    val coresPerExecutor = app.desc.coresPerExecutor
+    val minCoresPerExecutor = coresPerExecutor.getOrElse(1)
```

Comment:
hm, I thought we used `coresPerExecutor` down there, but maybe not in the latest code. If not we can merge these two.